src/mds/MDLog.cc

```diff
@@ -627,6 +627,13 @@ void MDLog::trim(int m)
 
   unsigned new_expiring_segments = 0;
 
+  unsigned max_expiring_segments = 0;
+  if (pre_segments_size > 0){
+    max_expiring_segments = max_segments/2;
+    assert(segments.size() >= pre_segments_size);
+    max_expiring_segments = std::max(max_expiring_segments,static_cast<unsigned>(segments.size() - pre_segments_size));
```

Comment:
std::max<unsigned>

otherwise, looks good

Reply:
update it. thanks.